app/services/benefit_check_service.rb

```diff
@@ -1,6 +1,19 @@
 class BenefitCheckService
   BENEFIT_CHECKER_NAMESPACE = 'https://lsc.gov.uk/benefitchecker/service/1.0/API_1.0_Check'.freeze
-  ApiError = Class.new(StandardError)
+  USE_MOCK = ActiveModel::Type::Boolean.new.cast(ENV['USE_BC_DEV_MOCK'])
+
+  class ApiError < StandardError
+    include Nesty::NestedError
+  end
+
+  def self.call(application)
+    return MockBenefitCheckService.call(application) if USE_MOCK && !Rails.env.production?
+
+    # :nocov:
+    # Not covered as this part of method will be skipped if USE_MOCK enabled
+    new(application).call
+    # :nocov:
```

Comment:
I don't understand why we can't cover this. Don't we have tests calling the real benefit checker but using VCR?

Reply:
If you look at the specs for `BenefitCheckService` they use VCR. The key thing to notice is that the instance tested is initiated directly via `BenefitCheckService.new`. So all the instance methods (including `#call`) are thoroughly tested using VCR.

All the other specs (for example from the controller) mock the call to the `call` method. To test a non-mocked version would require a new spec with a working VCR cassette. As I still don't have a working API connection from dev, there isn't a simple way of building one.

As the code `new(application).call` is already thoroughly tested, putting this call within a `:nocov:` seemed the simplest solution, and had minimal risks.

Reply:
You're right to query this. There is a simple way this can be avoided.

Reply:
Changing the test to use `BenefitCheckService.call(application)` instead of `BenefitCheckService.new` should do it.
I had a go at it: https://github.com/ministryofjustice/laa-apply-for-legal-aid/compare/add-dev-mock-for-benefit-checker...add-dev-mock-for-benefit-checker-julien?expand=1

Reply:
I found another way to do it - add a specific test of `BenefitCheckService.call` and mock the called to the instance method `call`. It ensures the behaviour is tested, with minimal change to the existing code.

Reply:
Up to you but It feels a bit like a shortcut, to be honest 😄 
If you were to re-write the `class BenefitCheckService` from scratch which approach would you choose?

Reply:
I'd do it the same way (that is build most of the specs using a instance initiated via `.new`) - but not mock the `BenefitCheckService.call` call in this latest spec. However, as I don't have a working API callout, I can't build the VCR cassette simply. So .... shortcut.